daemon/health.go

```diff
@@ -457,7 +457,7 @@ func getShell(cntr *container.Container) []string {
 	if runtime.GOOS != "windows" {
 		return []string{"/bin/sh", "-c"}
 	}
-	if cntr.OS != runtime.GOOS {
+	if cntr.ImagePlatform.OS != runtime.GOOS {
 		return []string{"/bin/sh", "-c"}
 	}
```

Comment:
Reminds me I still need to dig into some of these, I _think_ there's still some LCOW leftovers dangling around, and there may be some broken logic in some places (https://github.com/moby/moby/pull/42170)